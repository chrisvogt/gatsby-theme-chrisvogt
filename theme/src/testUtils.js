import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { ThemeProvider } from 'theme-ui'
import theme from './gatsby-plugin-theme-ui'

import store from './store'

export { store }

// React module that provides necessary context to components being tested.
export const TestProvider = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

export const TestProviderWithState = ({ children }) => (
  <ReduxProvider store={store}>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </ReduxProvider>
)
