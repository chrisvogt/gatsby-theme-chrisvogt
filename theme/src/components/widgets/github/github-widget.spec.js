import React from 'react'
import GitHubWidget from './github-widget'
import renderer from 'react-test-renderer'
import { TestProviderWithState } from '../../../testUtils'
import useSiteMetadata from '../../../hooks/use-site-metadata'

jest.mock('../../../hooks/use-site-metadata')

const mockSiteMetadata = {
  widgets: {
    github: {
      username: 'mockusername',
      widgetDataSource: 'https://fake-api.example.com/social/github'
    }
  }
}

describe('GitHub Widget', () => {
  beforeEach(() => {
    useSiteMetadata.mockImplementation(() => mockSiteMetadata)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('matches the loading state snapshot', () => {
    const tree = renderer
      .create(
        <TestProviderWithState>
          <GitHubWidget />
        </TestProviderWithState>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
