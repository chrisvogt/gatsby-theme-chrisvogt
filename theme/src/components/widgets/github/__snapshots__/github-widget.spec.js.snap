// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHub Widget matches the loading state snapshot 1`] = `
<section
  className="css-1pk2a5b"
  id="github"
>
  <header
    className="css-x6ikgs"
  >
    <h2
      className="css-1s4ek1q"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-github css-1f4q58s"
        data-icon="github"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      GitHub
    </h2>
    <div
      className="css-1rgtmy6"
    >
      <svg
        className="svg-loaders-svg css-ao632c"
        fill="#1E90FF"
        height="24"
        viewBox="0 0 135 140"
        width="24"
      >
        <rect
          height={120}
          rx={6}
          width={15}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={30}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={140}
          rx={6}
          width={15}
          x={60}
        >
          <animate
            attributeName="height"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={90}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={120}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
      </svg>
    </div>
  </header>
  <div
    className="css-1hw7zrq"
  />
  <div
    className="css-lnnjou"
  >
    <h3
      className="css-174rnww"
    >
      Pinned Items
    </h3>
    <p
      className="css-1qwovgy"
    >
      Pinned items on my GitHub profile.
    </p>
    <div
      className="css-r2v68e"
    >
      <a
        className="css-nf0qmy"
      >
        <div
          className="css-1rkcmha"
        >
          <div
            className="show-loading-animation"
          >
            <div
              className="css-z5fbhm"
            >
              <div>
                <div
                  className="rect-shape"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": 40,
                      "marginBottom": "2em",
                      "marginRight": 10,
                      "width": 40,
                    }
                  }
                />
              </div>
              <div
                className="css-fjakwv"
              >
                <div
                  className="text-block"
                  style={
                    {
                      "width": "100%",
                    }
                  }
                >
                  <div
                    className="text-row"
                    style={
                      {
                        "backgroundColor": "#efefef",
                        "height": "1em",
                        "marginTop": 0,
                        "maxHeight": "33.333333333333336%",
                        "width": "97%",
                      }
                    }
                  />
                  <div
                    className="text-row"
                    style={
                      {
                        "backgroundColor": "#efefef",
                        "height": "1em",
                        "marginTop": "0.7em",
                        "maxHeight": "33.333333333333336%",
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div
              className="text-block"
              style={
                {
                  "width": "100%",
                }
              }
            >
              <div
                className="text-row"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": "1em",
                    "marginTop": 0,
                    "maxHeight": "20%",
                    "width": "97%",
                  }
                }
              />
              <div
                className="text-row"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": "1em",
                    "marginTop": "0.7em",
                    "maxHeight": "20%",
                    "width": "100%",
                  }
                }
              />
              <div
                className="text-row"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": "1em",
                    "marginTop": "0.7em",
                    "maxHeight": "20%",
                    "width": "94%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </a>
      <a
        className="css-nf0qmy"
      >
        <div
          className="css-1rkcmha"
        >
          <div
            className="show-loading-animation"
          >
            <div
              className="css-z5fbhm"
            >
              <div>
                <div
                  className="rect-shape"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": 40,
                      "marginBottom": "2em",
                      "marginRight": 10,
                      "width": 40,
                    }
                  }
                />
              </div>
              <div
                className="css-fjakwv"
              >
                <div
                  className="text-block"
                  style={
                    {
                      "width": "100%",
                    }
                  }
                >
                  <div
                    className="text-row"
                    style={
                      {
                        "backgroundColor": "#efefef",
                        "height": "1em",
                        "marginTop": 0,
                        "maxHeight": "33.333333333333336%",
                        "width": "97%",
                      }
                    }
                  />
                  <div
                    className="text-row"
                    style={
                      {
                        "backgroundColor": "#efefef",
                        "height": "1em",
                        "marginTop": "0.7em",
                        "maxHeight": "33.333333333333336%",
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div
              className="text-block"
              style={
                {
                  "width": "100%",
                }
              }
            >
              <div
                className="text-row"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": "1em",
                    "marginTop": 0,
                    "maxHeight": "20%",
                    "width": "97%",
                  }
                }
              />
              <div
                className="text-row"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": "1em",
                    "marginTop": "0.7em",
                    "maxHeight": "20%",
                    "width": "100%",
                  }
                }
              />
              <div
                className="text-row"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": "1em",
                    "marginTop": "0.7em",
                    "maxHeight": "20%",
                    "width": "94%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className="css-1yz7e9k"
  >
    <h3
      className="css-uyljfj"
    >
      Last Pull Request
    </h3>
    <a
      className="css-1srnkuc"
    >
      <div
        className="css-dqblq9"
      >
        <div
          className="text-row show-loading-animation"
          style={
            {
              "backgroundColor": "#efefef",
              "height": "1em",
              "marginTop": 0,
              "maxHeight": undefined,
              "width": "100%",
            }
          }
        />
        <div
          className="css-b5lvlf"
        >
          <div
            className="text-row show-loading-animation"
            style={
              {
                "backgroundColor": "#efefef",
                "height": "15px",
                "marginTop": 0,
                "maxHeight": undefined,
                "width": "150px",
              }
            }
          />
        </div>
      </div>
    </a>
  </div>
</section>
`;
